src/main/java/step2/model/WinningLotto.java

```diff
@@ -1,23 +1,39 @@
 package step2.model;
 
 import java.util.List;
-import java.util.function.Predicate;
-import java.util.stream.Collectors;
 
 public class WinningLotto {
 
-	private List<Integer> winNumList;
+	private static final int SECOND_MATCH_NUM = 5;
+	private List<Integer> winLotto;
+	private int bonusNumber;
 
-	public WinningLotto(List<Integer> winNumList) {
-		this.winNumList = winNumList;
+
+	private WinningLotto(List<Integer> winLotto, int bonusNumber) {
+		this.winLotto = winLotto;
+		this.bonusNumber = bonusNumber;
+	}
+
+	public static WinningLotto of(List<Integer> winNumList, int bonusNumber) {
+		return new WinningLotto(winNumList, bonusNumber);
 	}
 
-	public int countMatch(List<Integer> lotto) {
-		return winNumList.stream()
-			.filter(
-				target -> lotto.stream().distinct().anyMatch(Predicate.isEqual(target)))
-			.collect(Collectors.toList())
-			.size();
+	public int countOfMatch(List<Integer> lotto) {
+		return (int) winLotto.stream()
+			.filter(lotto::contains)
+			.count();
 	}
 
+	public boolean isBonus(int countLottoMatch, List<Integer> lotto) {
+		long countOfBonus = 0;
+
+		if (countLottoMatch == SECOND_MATCH_NUM) {
+			countOfBonus = lotto.stream().filter(s -> s.equals(bonusNumber)).count();
+		}
+
+		if (countOfBonus > 0) {
+			return true;
+		}
+		return false;
```

Comment:
다음처럼 작성해볼 수 있을 것 같아요.
```suggestion
		return countOfBonus > 0;
```